src/envo/envo-edit.owl

```diff
@@ -3535,6 +3553,15 @@ Declaration(Class(<http://purl.obolibrary.org/obo/ENVO_03510038>))
 Declaration(Class(<http://purl.obolibrary.org/obo/ENVO_03510039>))
 Declaration(Class(<http://purl.obolibrary.org/obo/ENVO_03510040>))
 Declaration(Class(<http://purl.obolibrary.org/obo/ENVO_03510041>))
+Declaration(Class(<http://purl.obolibrary.org/obo/ENVO_03510055>))
+Declaration(Class(<http://purl.obolibrary.org/obo/ENVO_03510056>))
+Declaration(Class(<http://purl.obolibrary.org/obo/ENVO_03510057>))
```

Comment:
The definition and placement have no meaningful semantics regarding the composition of `permanent hair dye`. 
The textual definition is not sufficiently linked to the axioms either.

Looks like there should be a sub-hierarchy for "dyes", but that's more a role than anything metaphysical. 

Looking at the axioms, this is more (skipping better superclasses) "A liquid environmental material which is 1) manufactured, 2) composed primarily of hydrogen peroxide and ammonia 3) used to permanently alter hair colour" 

On 2) I don't know if these are the primary constituents, but you get the idea - we need differntia in the textual and axiomatic definitions focused on what the thing is, rather than its role.

On 3) - nothing permanently does anything - what is the actual standard threshold for a dye to be considered "permanent"?

Reply:
In references to the variety of terms that fall under the category of ink/dye/colorant etc. We have included two new subclasses of 'environmental material' called 'colorant fluid' and 'chemical colorant'. Both are considered manufactured. 
Below these two classes the following have become subclasses:
Under 'colorant fluid': 'pharmaceutical ink', 'gel ink', 'soy ink', ' and 'aqueous inkjet printer ink'
Under 'chemical colorant': 'permanent hair dye'

All definitions for these indicated subclasses have been updated to reflect their classification and composition with recommendations taken from your notes. Please advise any further revisions. Thanks!

Reply:
Thanks @laurenechan - 'colorant fluid' and chemical colorant still sound very role-based - I'll look at the latest file to see how this can work.